File: julia/run-length-encoding/run-length-encoding.jl
function encode(s)
   replace(s, r"(.)\1+" => function(s)
      string(length(s)) * s[1]
   end)
end

function decode(s)
   replace(s, r"(\d+)(\D)"=> function(s)
      repeat(s[end], parse(Int, s[1:end-1]))
   end)
end

File: kotlin/run-length-encoding/src/main/kotlin/RunLengthEncoding.kt
object RunLengthEncoding {

    fun encode(input: String): String =
        input.replace(Regex("(.)(\\1+)")) {
            String.format("%d%s", it.value.length, it.groupValues[1])
        }

    fun decode(input: String): String =
        input.replace(Regex("(\\d+)(.)")) {
            it.groupValues[2].repeat(it.groupValues[1].toInt())
        }
}
